Microsoft.Quantum.Katas/KataMagic.cs

```diff
@@ -0,0 +1,168 @@
+        /// The list of user-defined Q# code snippets from the notebook.
+        /// </summary>
+        protected ISnippets Snippets { get; }
+
+        /// <summary>
+        /// What this Magic does when triggered. It will:
+        /// - find the Kata to execute based on the Kata name,
+        /// - compile the code after found after the name as the user's answer.
+        /// - run (simulate) the kata.
+        /// </summary>
+        public virtual ExecutionResult Run(string input, IChannel channel)
+        {
+            channel = channel.WithNewLines();
+
+            // Expect exactly two arguments, the name of the Kata and the user's answer (code).
+            var args = input?.Split(new char[] { ' ', '\n', '\t' }, 2);
+            if (args == null || args.Length != 2) throw new InvalidOperationException("Invalid parameters. Usage: `%kata Test \"Q# operation\"`");
+
+            var name = args[0];
+            var code = args[1];
+
+            var kata = FindKata(name);
+            if (kata == null) throw new InvalidOperationException($"Invalid test name: {name}");
+
+            var userAnswer = Compile(code, channel);
+            if (userAnswer == null) { return ExecuteStatus.Error.ToExecutionResult(); }
+
+            return Simulate(kata, userAnswer, channel)
+                ? "成功 (success)!".ToExecutionResult()
+                : ExecuteStatus.Error.ToExecutionResult();
+        }
+
+        /// <summary>
+        /// Compiles the given code. Checks there is only one operation defined in the code,
+        /// and returns its corresponding OperationInfo
+        /// </summary>
+        public virtual OperationInfo Compile(string code, IChannel channel)
+        {
+            try
+            {
+                var result = Snippets.Compile(code);
+
+                foreach (var m in result.warnings) { channel.Stdout(m); }
+
+                // Gets the names of all the operations found for this snippet
+                var opsNames =
+                    result.Elements?
+                        .Where(e => e.IsQsCallable)
+                        .Select(e => e.ToFullName().WithoutNamespace(Microsoft.Quantum.IQSharp.Snippets.SNIPPETS_NAMESPACE))
+                        .OrderBy(o => o)
+                        .ToArray();
+
+                if (opsNames.Length > 1)
+                {
+                    channel.Stdout("Expecting only one Q# operation in code. Using the first one");
+                }
+
+                return Resolver.Resolve(opsNames.First());
+            }
+            catch (CompilationErrorsException c)
+            {
+                foreach (var m in c.Errors) channel.Stderr(m);
+                return null;
+            }
+            catch (Exception e)
+            {
+                channel.Stderr(e.Message);
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Executes the given kata using the provided `userAnswer` as the actual answer.
```

Comment:
C# XML comments don't support the `` ` `` syntax, so far as I'm aware. I think you need `<c>` and </c>` instead:
```suggestion
        /// Executes the given kata using the provided <c>userAnswer</c> as the actual answer.
```